### PR TITLE
Package samplerate.0.1.5

### PR DESCRIPTION
**Pull request by toots**
### `samplerate.0.1.5`
Samplerate audio conversion library
Bindings for the samplerate library which provides functions for changing samplerate of audio data

---
* Homepage: https://github.com/savonet/ocaml-samplerate
* Source repo: git+https://github.com/savonet/ocaml-samplerate.git
* Bug tracker: https://github.com/savonet/ocaml-samplerate/issues

---
:camel: Pull-request generated by opam-publish v2.0.2

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
The tests failed on OCaml < 4.08:
```
Processing  1/2: [samplerate: dune build]
+ /home/opam/.opam/4.07/bin/dune "build" "-p" "samplerate" "-j" "72" "@install" "@runtest" (CWD=/home/opam/.opam/4.07/.opam-switch/build/samplerate.0.1.5)
-     ocamlopt examples/test.exe (exit 2)
- (cd _build/default && /home/opam/.opam/4.07/bin/ocamlopt.opt -w -40 -g -o examples/test.exe src/samplerate.cmxa -I src examples/.test.eobjs/native/dune__exe__Test.cmx)
- File "_none_", line 1:
- Error: No implementations provided for the following modules:
-          Bigarray referenced from examples/.test.eobjs/native/dune__exe__Test.cmx
```

File: packages/conf-libsamplerate/conf-libsamplerate.1/opam [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+bug-reports: "https://github.com/ocaml/opam-repository/issues"
+authors: "libsamplerate dev team"
+license: "BSD"
+build: ["pkg-config" "--exists" "samplerate"]
+depends: [
+  "conf-pkg-config" {build}
+]
+depexts: [
+  ["libsamplerate-dev"] {os-family = "debian"}
+  ["libsamplerate-dev"] {os-family = "ubuntu"}
```
Comment (toots) [inline packages/conf-libsamplerate/conf-libsamplerate.1/opam:13]: Is this really needed?

File: packages/samplerate/samplerate.0.1.5/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+authors: ["The Savonet Team <savonet-users@lists.sourceforge.net>"]
+license: "LGPL-2.1"
+homepage: "https://github.com/savonet/ocaml-samplerate"
+bug-reports: "https://github.com/savonet/ocaml-samplerate/issues"
+depends: [
+  "dune" {> "2.0"}
+  "dune-configurator"
+  "conf-libsamplerate"
+  "conf-pkg-config"
+  "base-bigarray"
```
Comment (toots) [inline packages/samplerate/samplerate.0.1.5/opam:15]: @kit-ty-kate Just added the `base-bigarray`, let's hope this fixes the tests.

File: packages/conf-libsamplerate/conf-libsamplerate.1/opam (hunk 1 above)
Comment (kit-ty-kate) [reply packages/conf-libsamplerate/conf-libsamplerate.1/opam:13]: If we wish to support Ubuntu-derived distribution, yes.

Opam gets the value of `os-family` from the first value of `ID_LIKE` in `/etc/os-release`, which for Ubuntu-derivered distributions might be `ubuntu` first.

Comment (toots) [reply packages/conf-libsamplerate/conf-libsamplerate.1/opam:13]: Ha, tricky.

File: packages/conf-libsamplerate/conf-libsamplerate.1/opam [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  ["libsamplerate-devel"] {os-distribution = "centos"}
+  ["libsamplerate-devel"] {os-distribution = "fedora"}
+  ["libsamplerate-devel"] {os-family = "suse"}
+  ["samplerate-dev"] {os-family = "alpine"}
+  ["samplerate"] {os-distribution = "nixos"}
+  ["samplerate"] {os-family = "arch"}
+  ["libsamplerate"] {os = "freebsd"}
+  ["libsamplerate"] {os = "macos" & os-distribution = "homebrew"}
+]
+synopsis: "Virtual package relying on libFLAC"
```
Comment (toots) [inline packages/conf-libsamplerate/conf-libsamplerate.1/opam:23]:
```suggestion
synopsis: "Virtual package relying on libsamplerate"
```

**Comment by kit-ty-kate** [comment]
The same failure still appear in the tests:
```
#=== ERROR while compiling samplerate.0.1.5 ===================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/savonet/ocaml-samplerate/archive/v0.1.5.tar.gz)
# path                 ~/.opam/4.07/.opam-switch/build/samplerate.0.1.5
# command              ~/.opam/4.07/bin/dune build -p samplerate -j 72 @install @runtest
# exit-code            1
# env-file             ~/.opam/log/samplerate-29-ffb3fd.env
# output-file          ~/.opam/log/samplerate-29-ffb3fd.out
### output ###
#     ocamlopt examples/test.exe (exit 2)
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlopt.opt -w -40 -g -o examples/test.exe src/samplerate.cmxa -I src examples/.test.eobjs/native/dune__exe__Test.cmx)
# File "_none_", line 1:
# Error: No implementations provided for the following modules:
#          Bigarray referenced from examples/.test.eobjs/native/dune__exe__Test.cmx
```

**Comment by talex5** [comment]
I had this problem with capnproto on 4.07 too; see https://github.com/janestreet/base/issues/94.

**Comment by toots** [comment]
Just fixed the library's `dune` file to add `bigarray` to the required libraries, that should fix the problem.

**Comment by toots** [comment]
Looks like it's passing everywhere reasonable now..

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**